python/API_GW_REQUIRE_AUTHORIZER/API_GW_REQUIRE_AUTHORIZER.py

```diff
@@ -0,0 +1,505 @@
+except ImportError:
+    pass
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::::Account'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+#############
+# Main Code #
+#############
+
+_client_cache = None
+_config_rule_response = []
+API_GATEWAY_STAGE_TYPE = "AWS::ApiGateway::Stage"
+
+
+def _add_response(stage, status, event, comment):
+    global _config_rule_response
+
+    _config_rule_response.append(
+        build_evaluation(resource_id=stage, compliance_type=status, event=event, resource_type=API_GATEWAY_STAGE_TYPE,
+                         annotation=comment))
+    print(f"Reporting {stage} as {status}: {comment}")
+
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    """Form the evaluation(s) to be return to Config Rules
+
+    Return either:
+    None -- when no result needs to be displayed
+    a string -- either COMPLIANT, NON_COMPLIANT or NOT_APPLICABLE
+    a dictionary -- the evaluation dictionary, usually built by build_evaluation_from_config_item()
+    a list of dictionary -- a list of evaluation dictionary , usually built by build_evaluation()
+
+    Keyword arguments:
+    event -- the event variable given in the lambda handler
+    configuration_item -- the configurationItem dictionary in the invokingEvent
+    valid_rule_parameters -- the output of the evaluate_parameters() representing validated parameters of the Config Rule
+
+    Advanced Notes:
+    1 -- if a resource is deleted and generate a configuration change with ResourceDeleted status, the Boilerplate code will put a NOT_APPLICABLE on this resource automatically.
+    2 -- if a None or a list of dictionary is returned, the old evaluation(s) which are not returned in the new evaluation list are returned as NOT_APPLICABLE by the Boilerplate code
+    3 -- if None or an empty string, list or dict is returned, the Boilerplate code will put a "shadow" evaluation to feedback that the evaluation took place properly
+    """
+    config_rule_event = json.loads(event.get("invokingEvent"))
+
+    if valid_rule_parameters is not None:
+        valid_authorizers = valid_rule_parameters.get("RequireAuthTypes")
+
+    if "configurationItem" in config_rule_event:
+        configuration_item = config_rule_event.get("configurationItem")
+        api_id = configuration_item.get("resourceId").split("/")[2]
+        api_region = configuration_item.get("awsRegion")
+
+        global _client_cache
+        if _client_cache is None:
+            _client_cache = {api_region: get_client(service="apigateway", event=event, region=api_region)}
+
+        api_client = _client_cache.get(api_region)
+
+        # get all resources for this API
+        resources = api_client.get_resources(
```

Comment:
You must either rely solely on configuration_item or create a periodic rule. Otherwise the rule will not trigger on changes.

Reply:
In this case, the Stage is updated, and then I load the resources for the stage, which are not part of the configuration item. This would be the desired behaviour.